### ISSUE TITLE
Weird devices in /sys/class/hwmon break krun

### ISSUE THREAD

**Issue by cfbolz** [softdevteam/krun#245]
I have a strange entry in /sys/class/hwmon/hwmon1, which doesn't have a 'name' file. That breaks the temperature code. I worked around it to run tests with this kind of code, which just ignores such directories:

```
diff --git a/krun/platform.py b/krun/platform.py
index 9f4b08d..eba4ad5 100644
--- a/krun/platform.py
+++ b/krun/platform.py
@@ -686,8 +686,11 @@ class LinuxPlatform(UnixLikePlatform):
         hwmons = glob.glob(LinuxPlatform.HWMON_CHIPS_GLOB)
         for chip in hwmons:
             name_path = os.path.join(chip, "name")
-            with open(name_path) as fh:
-                chip_name = fh.read().strip()
+            try:
+                with open(name_path) as fh:
+                    chip_name = fh.read().strip()
+            except Exception:
+                continue

             inputs_glob = os.path.join(chip, "temp[0-9]*_input")
             chip_sensors = glob.glob(inputs_glob)
```

Any better ideas?

**Comment by vext01** [comment]
What else is in the directory? Is there anything we can use to identify the chip the sensors are on?

**Comment by cfbolz** [comment]
drwxr-xr-x 2 root root    0 Sep 16 14:14 power
-rw-r--r-- 1 root root 4.0K Sep 16 14:14 uevent
lrwxrwxrwx 1 root root    0 Sep 16 14:14 subsystem -> ../../../../../class/hwmon
lrwxrwxrwx 1 root root    0 Sep 16 14:14 device -> ../../../thinkpad_hwmon

**Comment by vext01** [comment]
Hrm, symlinks. presumably there are names in the symlinked dirs?

**Comment by cfbolz** [comment]
yes, in the ``device`` symlink

**Comment by vext01** [comment]
So we can either try to follow symlinks, or we can skip them as above.

Let's skip for now, and maybe later we can fix this properly. If you are going ahead with a PR, lease add a log message at warn level when we skip.

**Comment by vext01** [comment]
Is this fixed?

**Comment by ltratt** [comment]
I don't think we've done anything on this.

**Comment by ltratt** [comment]
It's not critical for the run on the main benchmarking machines, so I suggest we delay fixing it, and focus on more critical things.

**Comment by snim2** [comment]
I have the same problem, my `/sys/class/hwmon/hwmon2` is missing a `name` file.

**Comment by vext01** [comment]
If we can be 100% sure the chips attach in the same order each boot, we can use the path as an identifier, but I'm not sure there is such a guarantee.

(In OpenBSD each temperature sensor has a unique name)

**Comment by ltratt** [comment]
I don't think there's any such guarantee, at least not in general...

**Comment by vext01** [comment]
I also suspect that is the case.

**Comment by snim2** [comment]
OK, here's some info from my laptop:

```sh
$ ls /sys/class/hwmon/hwmon0/
name  power  subsystem  temp1_crit  temp1_input  uevent

$ ls /sys/class/hwmon/hwmon1/
name  power  subsystem  temp1_input  uevent

$ ls /sys/class/hwmon/hwmon2/
device  power  subsystem  uevent

$ ls /sys/class/hwmon/hwmon3/
name  power  subsystem  temp1_input  uevent

$ ls /sys/class/hwmon/hwmon4/
device  power      temp1_crit        temp1_input  temp1_max   temp2_crit_alarm  temp2_label  temp3_crit        temp3_input  temp3_max
name    subsystem  temp1_crit_alarm  temp1_label  temp2_crit  temp2_input       temp2_max    temp3_crit_alarm  temp3_label  uevent
```

`hwmon{0,1,2,3,4}` are all symlinks.

**Comment by snim2** [comment]
```sh
$ ls -al /sys/class/hwmon/hwmon2/
total 0
drwxr-xr-x 3 root root    0 Sep  6 17:36 .
drwxr-xr-x 3 root root    0 Sep  5 22:36 ..
lrwxrwxrwx 1 root root    0 Sep  6 17:33 device -> ../../../thinkpad_hwmon
drwxr-xr-x 2 root root    0 Sep  6 17:33 power
lrwxrwxrwx 1 root root    0 Sep  5 22:45 subsystem -> ../../../../../class/hwmon
-rw-r--r-- 1 root root 4096 Sep  5 22:36 uevent
```

and for contrast:

```sh
$ ls -al /sys/class/hwmon/hwmon0/
total 0
drwxr-xr-x 3 root root    0 Sep  6 17:36 .
drwxr-xr-x 5 root root    0 Sep  5 22:36 ..
-r--r--r-- 1 root root 4096 Sep  5 22:36 name
drwxr-xr-x 2 root root    0 Sep  6 17:33 power
lrwxrwxrwx 1 root root    0 Sep  5 22:36 subsystem -> ../../../../class/hwmon
-r--r--r-- 1 root root 4096 Sep  6 17:33 temp1_crit
-r--r--r-- 1 root root 4096 Sep  6 17:33 temp1_input
-rw-r--r-- 1 root root 4096 Sep  5 22:36 uevent
```

**Comment by vext01** [comment]
Here's Tom's info:

```
tfransham
:
5:50 PM
  

fsfod@debian:/work/krun/examples$ ls -la /sys/class/hwmon/
lrwxrwxrwx  1 root root 0 Sep  6 17:37 hwmon0 -> ../../devices/virtual/hwmon/hwmon0
lrwxrwxrwx  1 root root 0 Sep  6 17:37 hwmon1 -> ../../devices/platform/thinkpad_hwmon/hwmon/hwmon1
lrwxrwxrwx  1 root root 0 Sep  6 17:37 hwmon2 -> ../../devices/platform/coretemp.0/hwmon/hwmon2
fsfod@debian:/work/krun/examples$ ls -la /sys/class/hwmon/hwmon1/device
lrwxrwxrwx 1 root root 0 Sep  6 17:34 /sys/class/hwmon/hwmon1/device -> ../../../thinkpad_hwmon
fsfod@debian:/work/krun/examples$ ls -la /sys/class/hwmon/hwmon2/device
lrwxrwxrwx 1 root root 0 Sep  6 17:40 /sys/class/hwmon/hwmon2/device -> ../../../coretemp.0
```

**Comment by vext01** [comment]
Tom, could you paste the output of `find /sys/class/hwmon`? I need to see which chips have a name. Thanks.

**Comment by vext01** [comment]
Funnily enough, Linux docs say about hwmon interfaces:

```
All entries (except name) are optional
```

Great...

**Comment by vext01** [comment]
Fixed in #332
